### ISSUE TITLE
add content-length check when method is delete

### ISSUE THREAD

**Issue by finefun55** [fastify/fastify#936]
### better compatibility 
My service user added ```content-type=application/json``` for each request, even if the method is DELETE without body, of course, they got 400 error. 
I have added some codes to avoid the problem
```javascript
app.addHook("onRequest", (req, res, next) => {
  if (toLower(req.method) === "delete") {
    if (
      req.headers["content-type"] &&
      (req.headers["content-length"] === undefined || req.headers["content-length"] === "0")
    ) {
      delete req.headers["content-type"];
    }
  }
  next();
});
```
works fine, but i think that will be much better to  ignore content-type when content-length is undefined

```javascript
if (method === 'OPTIONS' || method === 'DELETE') {
    if (contentType === undefined || headers['content-length'] === undefined || headers['content-length'] === '0') {
      handler(reply)
    } else {
      context.contentTypeParser.run(contentType, handler, request, reply)
    }
    return
  }
```

**Comment by nwoltman** [comment]
Could you elaborate on how this would provide "better compatibility"? Is there a particular issue you're having?

Also for future reference, code examples are easier to read when they're inside a [highlighted code block](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code).

**Comment by nwoltman** [comment]
@finefun55 Thanks for fixing up the issue description. Sorry I didn't notice you had done that (GitHub only sends notifications for new comments, not edits).

This seems reasonable to me if the OPTIONS and DELETE cases are separated (since the [HTTP 1.1 spec](https://tools.ietf.org/html/rfc7231#section-4.3.7) says that for OPTIONS requests, the Content-Type header indicates that the request includes a body, so if there is a Content-Type header, then the body should be parsed).

**Comment by delvedor** [comment]
Closed in #1027
